### PR TITLE
feat(editor): Improve how we show default Agent prompt and Memory session parameters

### PR DESCRIPTION
**Pull request by OlegIvaniv**
## Summary

Currently, our AI nodes default to using expressions like `{{ $json.chatInput }}` and `{{ $json.sessionId }}` for certain fields. However, this implementation causes confusion to the new users.

This PR improves the UX by:
1. Making these default expression fields visible but disabled when in automatic modes (e.g., when `promptType` is `auto`)
2. Introducing a new `disabledOptions` property that follows the same logic as `displayOptions` but disables fields instead of hiding them

## Changes

- Added `disabledOptions` interface parallel to `displayOptions` for node properties
- Updated parameter input components to handle disabled state
- Modified expression editor components to support read-only mode styling
- Added version checks to ensure backward compatibility
- Applied the changes to:
  - Chain LLM node
  - Chain Retrieval QA node
  - SQL Agent node
  - Various memory nodes (Buffer Window, Motorhead, Postgres Chat, Redis Chat, Xata, Zep)
  - OpenAI Assistant node

https://github.com/user-attachments/assets/84e48391-c881-43f8-b858-33e4f91a5f00

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

**Review by CharlieKolb — COMMENTED** [review]
The code looks good and the feature looks even better!

I could see some users not immediately getting that/why the field is disabled, having a tooltip on hover that points calls out the controlling field or a tag next to the parameter name might be a good follow up item.
Having a visible hierarchy with indentation would also help.

Please do get another opinion as well since I'm new and mostly unfamiliar with existing best practices, as my comments will prove :D

File: packages/@n8n/nodes-langchain/nodes/chains/ChainLLM/ChainLlm.node.ts [hunk 1]
```diff
@@ -315,7 +315,7 @@ export class ChainLlm implements INodeType {
 				},
 			},
 			{
-				displayName: 'Prompt',
+				displayName: 'Prompt Source',
```
[the remark this answers is not in the document]
Comment (OlegIvaniv) [reply packages/@n8n/nodes-langchain/nodes/chains/ChainLLM/ChainLlm.node.ts]: Good point, I updated all of them to use `promptTypeOptions` from the shared utils file

File: packages/@n8n/nodes-langchain/nodes/agents/Agent/Agent.node.ts [hunk 2]
```diff
@@ -341,6 +341,25 @@ export class Agent implements INodeType {
 					},
 				},
 			},
+			{
+				displayName: 'Text From Previous Node',
```
[the remark this answers is not in the document]
Comment (OlegIvaniv) [reply packages/@n8n/nodes-langchain/nodes/agents/Agent/Agent.node.ts]: I consolidated all of them to use the shared export from the utils

**Comment by OlegIvaniv** [comment]
> The code looks good and the feature looks even better!
> 
> I could see some users not immediately getting that/why the field is disabled, having a tooltip on hover that points calls out the controlling field or a tag next to the parameter name might be a good follow up item. Having a visible hierarchy with indentation would also help.
> 
> Please do get another opinion as well since I'm new and mostly unfamiliar with existing best practices, as my comments will prove :D

Appreciate the review, @CharlieKolb! I updated the PR to use shared `textFromPreviousNode` and `promptTypeOptions` everywhere. Could you have another look, please?

**Review by CharlieKolb — APPROVED** [review]
Thanks for the example workflow, that's great :D 

PR looks good to go from my side 👍

**Review by elsmr — APPROVED** [review]
LGTM! 🎉 

Nice work, I like how you kept `disabledOptions` simple by reusing the `displayOptions` format. Maybe some of that code should be renamed in the future.

`show` and `hide` don't make much sense in disabledOptions. It's becoming a more general "Condition" object instead of something specific for hiding parameters.

**Comment by OlegIvaniv** [comment]
> LGTM! 🎉
> 
> Nice work, I like how you kept `disabledOptions` simple by reusing the `displayOptions` format. Maybe some of that code should be renamed in the future.
> 
> `show` and `hide` don't make much sense in disabledOptions. It's becoming a more general "Condition" object instead of something specific for hiding parameters.

Thanks! 
+1 on renaming it later to something more general(`conditionalParametersOptions`?), I would also love to use it for specifying the default value based on condition, but that breaks all kinds of stuff right now :D

**Comment by janober** [comment]
Got released with [`n8n@1.68.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.68.0)

**Comment by rogaha** [comment]
ScreenFloat Shot of Google Chrome at Nov 20, 2024 at 11
<img width=
_30_52 PM" src="https://github.com/user-attachments/assets/9cf5ebf8-74da-4fd4-8c7d-922811f9b041">

This feature created a `bug` in the `OpenAI` nodes.

**Comment by OlegIvaniv** [comment]
Thanks for that @rogaha 🙏  Pushed a fix, it'll go with the next release

**Comment by rogaha** [comment]
Awesome, thanks @OlegIvaniv for the quick fix.

**Pull request merged**